lib/puppet/provider/oneview_volume_template/c7000.rb

```diff
@@ -14,35 +14,15 @@
 # limitations under the License.
 ################################################################################
 
-require_relative '../login'
-require_relative '../common'
-require 'oneview-sdk'
+require_relative '../oneview_resource'
 
-Puppet::Type.type(:oneview_volume_template).provide(:oneview_volume_template) do
-  mk_resource_methods
+Puppet::Type::Oneview_volume_template.provide :c7000, parent: Puppet::OneviewResource do
+  desc 'Provider for OneView Volume Template using the C7000 variant of the OneView API'
 
-  def initialize(*args)
-    super(*args)
-    @client = OneviewSDK::Client.new(login)
-    @resourcetype = OneviewSDK::VolumeTemplate
-    # Initializes the data so it is parsed only on exists and accessible throughout the methods
-    # This is not set here due to the 'resources' variable not being accessible in initialize
-    @data = {}
-  end
+  confine true: login[:hardware_variant] == 'C7000'
 
-  def self.instances
-    @client = OneviewSDK::Client.new(login)
-    matches = OneviewSDK::VolumeTemplate.find_by(@client, {})
-    matches.collect do |line|
-      name = line['name']
-      data = line.inspect
-      new(name: name,
-          ensure: :present,
-          data: data)
-    end
-  end
+  mk_resource_methods
 
-  # Provider methods
   def exists?
```

Comment:
This method could probably be removed. The parent class appears to do the same as this.